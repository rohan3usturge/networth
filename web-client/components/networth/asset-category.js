import { Asset } from "./asset";

const AssetCategory = ({ category, onChange }) => {
  const handleChange = (changedItem) => {
    const changed = { ...category };
    const newItems = category.items.map((item) => {
      if (item.lineItem.id === changedItem.lineItem.id) {
        return changedItem;
      }
      return item;
    });
    changed.items = newItems;
    if (onChange) {
      onChange(changed);
    }
  };

  return (
    <table className="table table-borderless">
      <thead>
        <tr className="d-flex">
          <th scope="col" className="col-4">
            {category.meta.name}
          </th>
          <th className="col-8"></th>
        </tr>
      </thead>
      <tbody>
        {category.items.map((item) => {
          return (
            <Asset
              key={item.lineItem.id}
              asset={item}
              onChange={handleChange}
            />
          );
        })}
      </tbody>
    </table>
  );
};

export { AssetCategory };
