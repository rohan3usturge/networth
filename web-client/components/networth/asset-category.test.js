import { shallow } from "enzyme";
import { AssetCategory } from "./asset-category";

describe("AssetCategory", () => {
  it("should render properly and match snapshot", () => {
    const wrapper = shallow(
      <AssetCategory
        category={{ meta: {}, items: [] }}
        currency="CAD"
        disabled={true}
        onChange={() => {}}
        key="2323"
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
