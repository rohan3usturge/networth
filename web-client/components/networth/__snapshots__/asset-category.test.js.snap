// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AssetCategory should render properly and match snapshot 1`] = `
<table
  className="table table-borderless table-sm"
>
  <thead>
    <tr>
      <th
        scope="col"
      />
      <th />
    </tr>
  </thead>
  <tbody />
</table>
`;
